src/App.js

```diff
@@ -11,44 +26,52 @@ class App extends React.Component {
       currWord: getRandomWord(),
       // guessedLetters stores all letters a user has guessed so far
       guessedLetters: [],
-      // Insert num guesses left state here
-      // Insert form input state here
+      numGuess: 10,
+      userWord: "",
+      result: "",
+      score: 0,
     };
   }
 
-  generateWordDisplay = () => {
-    const wordDisplay = [];
-    // for...of is a string and array iterator that does not use index
-    for (let letter of this.state.currWord) {
-      if (this.state.guessedLetters.includes(letter)) {
-        wordDisplay.push(letter);
-      } else {
-        wordDisplay.push("_");
-      }
-    }
-    return wordDisplay.toString();
+  reset = () => {
+    this.setState({
+      currWord: getRandomWord(),
+      guessedLetters: [],
+      numGuess: 10,
+      result: "",
+    });
   };
 
-  // Insert form callback functions handleChange and handleSubmit here
+  updateGuess = (userWord) => {
+    const updateGuess = [...this.state.guessedLetters, userWord];
+    const updateNumGuess = this.state.numGuess - 1;
+    const result = detResult(this.state.currWord, updateGuess, updateNumGuess);
+    result === "win" && this.setState({ score: this.state.score + 1 });
+    this.setState({
+      guessedLetters: updateGuess,
+      numGuess: updateNumGuess,
+      result: result,
+    });
```

Comment:
We can use optional object properties here

```suggestion
    this.setState({
      guessedLetters: updateGuess,
      numGuess: updateNumGuess,
      result: result,
      ...(result === "win" && { score: this.state.score + 1 })
    });
```